File: python/sort/insert.py
# -*- coding: utf-8 -*-
from sort import AbstractSort


class InsertSort(AbstractSort):
    @classmethod
    def sort(cls, origin):
        length = len(origin)
        for i in range(1, length):
            for j in range(i, 0, -1):
                if origin[j] < origin[j - 1]:
                    tmp = origin[j]
                    origin[j] = origin[j - 1]
                    origin[j - 1] = tmp

File: php/sort/Insert.php
<?php
require ("AbstractSort.php");
class Insert extends AbstractSort
{
	public static function sort(array $insert)
	{
		$length = count($insert);
		for ($i=1; $i < $length; $i++) {
			for ($j=$i; $j > 0; $j--) {
				if ($insert[$j] < $insert[$j-1]) {
					$tmp = $insert[$j];
					$insert[$j] = $insert[$j - 1];
					$insert[$j - 1] = $tmp;
				}
			}
		}
		return $insert;
	}

}
